src/vconsole/vconsole-setup.c

```diff
@@ -47,43 +47,39 @@
 
 #define MAX_CONSOLES 63
 
-static bool is_vconsole(int fd) {
-        unsigned char data[1];
+static int is_vconsole(int fd) {
+        unsigned char subcode;
 
-        data[0] = TIOCL_GETFGCONSOLE;
-        return ioctl(fd, TIOCLINUX, data) >= 0;
+        subcode = TIOCL_GETFGCONSOLE;
+        return ioctl(fd, TIOCLINUX, &subcode);
 }
 
-static bool is_allocated(int fd, int idx) {
+static int is_allocated(int fd, int idx) {
         struct vt_stat vcs = {};
         char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
 
         if (fd >= 0) {
-                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
-                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
-                        return false;
-                }
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0)
+                        return log_warning_errno(errno, "VT_GETSTATE failed: %m");
                 idx = vcs.v_active;
         }
 
         xsprintf(vcname, "/dev/vcs%i", idx);
-        if (access(vcname, F_OK) < 0)
-                return false;
-        return true;
+        return access(vcname, F_OK);
```

Comment:
No, that's wrong. Your function now returns non-zero on error, and 0 if the terminal is actually allocated. In this case we don't care about the error value, so it's probably better to keep it returning `bool` as before.